symfit/contrib/interactive_guess/examples/simple_3D_example.py

```diff
@@ -0,0 +1,31 @@
+# -*- coding: utf-8 -*-
+
+from symfit import variables, Parameter, exp, Fit, Model
+from symfit.distributions import Gaussian
+from symfit.contrib.interactive_guess import InteractiveGuess
+import numpy as np
+
+
+x, y, z = variables('x, y, z')
+mu_x = Parameter('mu_x', 10)
+mu_y = Parameter('mu_y', 10)
+sig_x = Parameter('sig_x', 1)
+sig_y = Parameter('sig_y', 1)
+
+
+
+model = Model({z: Gaussian(x, mu_x, sig_x) * Gaussian(y, mu_y, sig_y)})
+x_data = np.linspace(0, 25, 50)
+y_data = np.linspace(0, 25, 50)
+x_data, y_data = np.meshgrid(x_data, y_data)
+x_data = x_data.flatten()
```

Comment:
Why does the user have to flatten? I think that is a bit unnatural, personally as a user I'm very fearful of flatten so I would never think of typing this.

Reply:
Without the flatten it raises an error, but I don't remember from where. But I also don't agree with the analysis, if you plug in a 2d array for x and y, z will also be 2d for this model. And the reason I'm always fearful of using flatten is because it could lead to contracting the wrong shapes so long as the flattened array is the same size, whereas without flatten you will get a healthy broadcasting error if the shapes are not alligned.

Reply:
Hmmn, ok. I see your point. Still though, the model does `f(R, R) -> R`, not `f(R^2, R^2) -> R^2`. I'll try to hunt down where this goes wrong, and whether I can circumvent it.
What I want/need to do is for every independent variable, create `n` linearly spaced points, and evaluate the model on those.
At the same time, I need to make a contourplot of the given `z` data for every projection. This means that I need to do this:
```
projected_data = np.column_stack((x_data, z_data)).T
kde = gaussian_kde(projected_data)
```
and also for `(y_data, z_data)`. And `projected_data`has to be 1 or 2D: https://docs.scipy.org/doc/scipy-0.15.1/reference/generated/scipy.stats.gaussian_kde.html

So no, there's no way around flattening the data, and in that case I'd rather let the user do it.

Reply:
In addition, it's needed for plotting.